### PR TITLE
Allow configuring RKE ACI network provider

### PR DESCRIPTION
**Pull request by fpants**
Issue: https://github.com/rancher/terraform-provider-rancher2/issues/922

This PR adds logic for configuring RKE clusters with the ACI network provider.

The Rancher API already supports this (see https://<rancher-host>/v3/schemas/aciNetworkProvider), so this is only to also expose this in the Rancher Terraform provider.

Similar ACI configuration is already available in the RKE Terraform provider: https://registry.terraform.io/providers/rancher/rke/latest/docs/resources/cluster#aci_network_provider

I'm not entirely sure everything is covered, so please make suggestions for any improvements.

### CODE REVIEW

File: rancher2/schema_cluster_rke_config_network.go
Comment (a-blender) [inline rancher2/schema_cluster_rke_config_network.go:33]: The following fields need to be required/sensitive/optional as specified in the [rke_cluster](https://registry.terraform.io/providers/rancher/rke/latest/docs/resources/cluster) resource docs. To make a field Required/Sensitive, add `Sensitive: true.` We want to make sure that if a user specifies an `aci_network_provider` configuration, the necessary fields are there to prevent provisioning errors when rke is deployed.

Required/Sensitive: `token, apic_user_key, apic_user_crt`

Required: `system_id, apic_hosts, apic_user_name, encap_type, mcast_range_start, mcast_range_end, aep, vrf_name, vrf_tenant, l3out, node_subnet, l3out_external_networks, extern_dyanmic, extern_static, node_svc_subnet, kube_api_vlan, service_vlan`

Optional: `snat_port_range_start, snat_port_range_end, snat_ports_per_node`

File: docs/resources/cluster.md
Comment (a-blender) [inline docs/resources/cluster.md]: Update args to Required/Sensitive or Required to match the terraform code

**Review by a-blender — CHANGES_REQUESTED** [review]
Please update the following

**Comment by fpants** [comment]
> Please update the following, otherwise looks great!

Thanks! I've updated the fields and the corresponding docs.

File: docs/resources/cluster.md
Comment (a-blender) [inline docs/resources/cluster.md:1019]: I don't see `apic_refresh_time` in the rke docs https://registry.terraform.io/providers/rancher/rke/latest/docs/resources/cluster. It doesn't appear to be supported by RKE, why was this added?

File: docs/resources/cluster.md
Comment (a-blender) [inline docs/resources/cluster.md:1024]: Same comment as above for `capic, controller_log_level, and drop_log_enable`

**Review by a-blender — CHANGES_REQUESTED** [review]
Thanks for the changes! I see a whole bunch of fields here not listed in the `rke_docs.` Could you explain why they are added and if they aren't needed, remove them?

File: docs/resources/cluster.md
Comment (fpants) [reply docs/resources/cluster.md:1019]: I copied them straight from the Rancher API schema:
https://<rancher-instance>/v3/schemas/aciNetworkProvider
So it's a 1:1 between the Rancher API and the Terraform provider.

I've used the API, and the extra fields (compared to the RKE provider) aren't required for the ACI CNI to function, but I thought it could be useful for others.

Comment (a-blender) [reply docs/resources/cluster.md:1019]: That isn't working for me. Could you provide another link to the schema so I can check it out?

Comment (fpants) [reply docs/resources/cluster.md:1019]: Sorry, it seems markdown removed parts of the URL (I was trying to show a local Rancher instance). I found the schema in the Rancher repo: https://github.com/rancher/rancher/blob/release/v2.6/pkg/client/generated/management/v3/zz_generated_aci_network_provider.go

**Review by a-blender — COMMENTED** [review]
LGTM, fields wise. We may need end-to-end acceptance tests for a basic implementation of the `aci_network_provider` field in the tf config for RKE clusters using the rancher2 provider. Merging this PR is pending some internal discussions on our end. Thank you for your patience.

File: docs/resources/cluster.md
Comment (a-blender) [reply docs/resources/cluster.md:1019]: Thank you, that's what I needed

**Comment by fpants** [comment]
Thanks! We have environments running with Rancher managing RKE clusters using the ACI CNI provider, and provisioning them using the Terraform provider with this code is working. I'd be happy to assist with end-to-end testing.

**Comment by a-blender** [comment]
Blocked by https://github.com/rancher/kontainer-driver-metadata/pull/922. Waiting on all feedback comments to be addressed and tested before reviewing with another tf SME. Once that PR is merged then this one can be re-reviewed and merged for ACI support in the Terraform provider

**Comment by a-blender** [comment]
ACI support https://github.com/rancher/kontainer-driver-metadata/pull/922 (KDM) and https://github.com/rancher/rke/pull/2963 (rke) have been merged, unblocking

**Review by a-blender — COMMENTED** [review]
Please add tf structure tests for `aci_network_provider` fields to make sure expansion and flatteners work correctly. Also provision an RKE cluster and import it into rancher with a simple ACI configuration to make sure that works. Then this will good to merge

**Comment by a-blender** [comment]
Apologies! Tests have already been added

The rke PR https://github.com/rancher/rke/pull/2963 has added extra fields to CNI version 5.2.3.2 here https://github.com/rancher/rke/pull/2963/files#diff-d6d39102a5c2432600e3ffac3abe79885d842926704befe5bdb3063be71518b7R653-R663. Please add these fields to this PR and update the tests, thank you

**Comment by fpants** [comment]
Before I can add the fields, I think they also need to be added here: https://github.com/rancher/rancher/blob/release/v2.6/pkg/client/generated/management/v3/zz_generated_aci_network_provider.go
Otherwise they can't be referenced and result in compile errors like:
```
rancher2/structure_cluster_rke_config_network.go:21:12: in.ApicRefreshTickerAdjust undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method ApicRefreshTickerAdjust)
```
I'm not sure how those files are generated in the rancher/rancher repo, but according to the [latest commit to the file](https://github.com/rancher/rancher/commit/81a68a765c6aafeb5bc8d2aff7587f03a878f29e), it's based on the rancher/rke go dependency. Perhaps the rke package needs to be released first?

**Comment by a-blender** [comment]
@fpants Correct, the tf rancher2 provider uses rancher, which uses rke under the hood. The ACI field support is not in a released version of rke yet so rancher doesn't know about them. Go ahead and add the fields/tests to this PR. I'll make sure the latest version of rke is released before this PR gets merged.

Merge blocked by rke v1.3.14 release https://github.com/rancher/rke/releases

**Comment by fpants** [comment]
Thanks @annablender, that makes sense. I have added the fields introduced in ACI-CNI 5.2.3.2.
I have also tested importing an existing RKE cluster with the ACI CNI installed, and it shows up and works in Rancher without issues.

**Comment by a-blender** [comment]
@fpants Thank you!

I see the build is failing
```
# github.com/rancher/terraform-provider-rancher2/rancher2 [github.com/rancher/terraform-provider-rancher2/rancher2.test]
112s
358	rancher2/structure_cluster_rke_config_network.go:21:11: in.ApicRefreshTickerAdjust undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method ApicRefreshTickerAdjust)
112s
359	rancher2/structure_cluster_rke_config_network.go:22:41: in.ApicRefreshTickerAdjust undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method ApicRefreshTickerAdjust)
112s
360	rancher2/structure_cluster_rke_config_network.go:27:11: in.ApicSubscriptionDelay undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method ApicSubscriptionDelay)
112s
361	rancher2/structure_cluster_rke_config_network.go:28:38: in.ApicSubscriptionDelay undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method ApicSubscriptionDelay)
112s
362	rancher2/structure_cluster_rke_config_network.go:45:11: in.DisablePeriodicSnatGlobalInfoSync undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method DisablePeriodicSnatGlobalInfoSync)
112s
363	rancher2/structure_cluster_rke_config_network.go:46:53: in.DisablePeriodicSnatGlobalInfoSync undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method DisablePeriodicSnatGlobalInfoSync)
112s
364	rancher2/structure_cluster_rke_config_network.go:48:11: in.DisableWaitForNetwork undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method DisableWaitForNetwork)
112s
365	rancher2/structure_cluster_rke_config_network.go:49:39: in.DisableWaitForNetwork undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method DisableWaitForNetwork)
112s
366	rancher2/structure_cluster_rke_config_network.go:54:11: in.DurationWaitForNetwork undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method DurationWaitForNetwork)
112s
367	rancher2/structure_cluster_rke_config_network.go:55:40: in.DurationWaitForNetwork undefined (type *"github.com/rancher/rancher/pkg/client/generated/management/v3".AciNetworkProvider has no field or method DurationWaitForNetwork)
112s
368	rancher2/structure_cluster_rke_config_network.go:55:40: too many errors
112s
369	FAIL	github.com/rancher/terraform-provider-rancher2/rancher2 [build failed]
128s
370	FAIL
```

This is happening because rancher does not have the additional ACI fields supported in the latest rke in its types structure. We are releasing rke `v1.3.13-rc4` https://github.com/rancher/rancher/pull/38295 by today. Once the CI passes here this will be ready for a final re-review and merge.

**Comment by a-blender** [comment]
RKE 1.3.14 has been merged https://github.com/rancher/rancher/pull/38295 and started being used by rancher on July 15, which should fix the above types error. Dependencies also needed to be bumped. The PR for that https://github.com/rancher/terraform-provider-rancher2/pull/959 has also been merged. Waiting on CI... will ping reviewers when this passes

**Review by a-blender — APPROVED** [review]

**Review by HarrisonWAffel — APPROVED** [review]
LGTM, build is passing

**Comment by annablender** [comment]
@fpants Could you possibly attach the RKE config for the imported cluster that you tested with so we can test on our end as well? Thank you

**Pull request merged**
